### DISCUSSION TITLE
Is it possible to build only the local tegraflash binaries without also building Linux OR target a different archetecture for tegraflash?

### DISCUSSION THREAD

**Discussion by conradmenchine** [OE4T/meta-tegra#1574] [Q&A]
Hi,
I'm interested in building just the local tegraflash binaries for a lab machine (Raspberry pi). From what I've seen, this only really happens as part of a full build. Is there a way to build just this portion? Or alternatively, is there a way to target a different architecture for these binaries? Thanks

**Comment by madisongh** [comment]
Which binaries are you asking about?  My guess is it's the host-side ones for initiating the flashing, since the target-side ones are meant specifically for the Jetson hardware only.  Unfortunately, we don't have sources for NVIDIA's host-side flashing tools.  They come in pre-built form only, and are x86_64 binaries.

You could try enabling binfmt support on your RPi's Linux to run them. I've run builds on ARM64 servers using that approach, and it's worked OK for build-time use of those tools. But I have no idea whether they'd run well enough, particularly on a small system like the RPi, to successfully perform the flashing operations.

**Comment by dwalkes** [comment]
There was some work done a while back on this and an example sequence at https://github.com/OE4T/meta-tegra/issues/480#issuecomment-725604856 which worked with tegraflash on RPI once upon a time.

**Comment by conradmenchine** [comment]
Thank you for the response! I didn't realize that these were pre-compiled binaries from Nvidia. The QEMU option that you linked looks promising though, so I'll try that.
